docs/learn/smart-contract-developers/create-lsp7-token.md

```diff
@@ -108,55 +125,78 @@ The `privateKey` coming from your UP extension is the private key of the EOA tha
 :::
 Create the script that will deploy the contract as your Universal Profile.
 
-1. Create a `ethers.wallet` instance with your private key (the `signer`)
+1. Add your Universal Profile address to your `.env` file in `UP_ADDR`
 2. Load the associated UP
 3. Get the bytecode of your contract
-4. use `staticCall` method to get the address of the contract
-5. deploy the contract
+4. Use `staticCall` method to get the address of the contract
+5. Deploy the contract
 
 ```ts title="scripts/deployUP.ts"
-import hre from 'hardhat';
 import { ethers } from 'hardhat';
 import * as dotenv from 'dotenv';
-import UniversalProfile from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725Account.json';
+import UniversalProfileArtifact from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725Account.json';
+import { CustomToken__factory } from '../typechain-types';
 
 // load env vars
 dotenv.config();
-const { UP_ADDR, PRIVATE_KEY } = process.env;
+const { UP_ADDR } = process.env;
 
 async function main() {
-  // setup provider
-  const provider = new ethers.JsonRpcProvider(
-    'https://rpc.testnet.lukso.network',
-  );
-  // setup signer (the browser extension controller)
-  const signer = new ethers.Wallet(PRIVATE_KEY as string, provider);
   // load the associated UP
-  const UP = await ethers.getContractAt('UniversalProfile', UP_ADDR as string);
-  console.log('🔑 EOA: ', signer.address);
-  console.log('🆙 Universal Profile: ', await UP.getAddress());
+  const universalProfile = await ethers.getContractAtFromArtifact(
```

Comment:
Why was it needed to change the method here from `getContractAt` to `getContractAtFromArtifact`?

Reply:
> For the following guide, we are usin g... `ethers v6` ...